### DISCUSSION TITLE
Setting expectations for community review

### DISCUSSION THREAD

**Discussion by ahwagner** [HGVSnomenclature/hgvs-nomenclature#107] [HVNC operations]
In https://github.com/HGVSnomenclature/hgvs-nomenclature/discussions/17#discussioncomment-7701850, @ifokkema wrote:
>A large number of pull requests have been merged without proper review in the last few weeks

While the large number of changes that have been done to fully overhaul the website ahead of the `24.1.0` release are–by necessity–being done without an extensive community review for each PR, I think it would be good for us to provide expectations to HGVS stakeholders about how long issues should take to be addressed and resolved. This includes expectations for:

- how long the committee will deliberate on a request before providing a decision / other feedback to community proposal submitters
- who is responsible for writing a community proposal page
- who is responsible for implementing a new feature in response to a proposal
- what our versions mean (e.g. `24.1.1` means the second (`.1`) version of the spec released in January 2024; not a version of the spec released on January 1 2024)
- what additional files (e.g. change logs) need to be maintained with Pull Requests
- how long specification maintainers will leave Pull Requests open for review
- who is required to approve Pull Requests before they are merged

There are probably other areas where we can improve transparency with the community, but I think these bullets would make for a good start and help keep us collectively accountable to an agreed set of expectations.

**Comment by ifokkema** [comment]
These are all good points, although I just meant that PRs were merged shortly after creation before a review from one of the other code owners was possible.

About the versioning; note that currently, according to the current website, the version won't contain any date information, even though version 24 was chosen because of the year. I find this confusing and prefer your examples, but this also shows that some decisions (although perhaps considered minor) have been made without discussion, even between all the code owners. It can sometimes be effective to make quick decisions, but now it feels like some changes were made in a rush.

**Comment by reece** [comment]
I agree with all of the above -- once the site stabilizes, changes should have a high bar and be reviewed.

However, there's no way this project would have happened by consensus or by asking for deliberation on every decision. I've made 339 commits and written lots of scripts to make this work. The next nearest contributor is at 27 commits. All I mean by this is that just by numbers, I have a lot more opportunity to make mistakes and make decisions that perturb people.

@ifokkema: I admit to having been on the fence about the version number. If you want something else, propose it.

**Reply by ifokkema** [reply]
I don't think anybody disagrees that you have made this happen and have done, by far, most of the work.
I'm simply pointing to our methods; mistakes can be prevented by having PRs reviewed. Decisions that are not regarding the content and therefore don't require the entire group can be taken in a smaller but more active group. I believe this work will benefit from these steps. If these steps bring issues (slowdown, etc), then we could look at how to solve those issues.


> @ifokkema: I admit to having been on the fence about the version number. If you want something else, propose it.

I'm observing that the current choice for versioning is a mix of semantic versioning with a once-off applied date-based version. I believe this is confusing. I believe we should stick to either semantic versioning or date-based versioning. I have a preference for date-based versioning:
- It is what was used before, so we won't have to change anything;
- I don't see the benefit of changing it to semantic versioning as this is not software;
- As I feel we're handling data/guidelines and not software, I fear we'll either end up having discussions on whether a change is backward-incompatible, or we'll just never even reach version 25.

**Reply by jfjlaros** [reply]
This is my stance on the matter of versioning: https://github.com/HGVSnomenclature/hgvs-nomenclature/discussions/12#discussioncomment-4552479
